Updating content that storage is always partitioned

HannahPeuckmann opened the pull request:
### Description

Updating the outdated documents that deal with Storage Access API/ storage access grants. Storage is always partitioned and storage access grants are limited to access to cookies now.

### Motivation

The documentation about the Storage Access API and automatic storage access grants are outdated since the implementation changed in Fx109. The outdated documentation led to confusion because developers expected to see the old behavior. 

### Additional details

Release notes: https://www.mozilla.org/en-US/firefox/109.0/releasenotes/
Meta bug: [Bug 1758737 - [meta] Always Partition Storage](https://bugzilla.mozilla.org/show_bug.cgi?id=1758737)

### Related issues and pull requests

Commit:
Updating content that storage is always partitioned and storage access grants are limited to cookies

Commit:
Updating content that storage is always partitioned and storage access grants are limited to cookies

Commit:
Avoiding the phrase 'cookie storage'

Commit:
Avoiding the phrase 'cookie storage'

Trikolon reviewed (commented):
Thanks for the MDN updates! This will make storage access a bit less confusing I hope.
Your patch changes a couple of mentions of storage to "cookies". While I don't think that's strictly required (the spec still talks about storage and cookies being the only "storage" affected by the storage access api), but it might help developers understand better that we only mean cookies.

I can't officially r+ or r- on this repo so just leaving some comments.

Commit:
implemented review from Tricolon

Commit:
implemented review from Tricolon

HannahPeuckmann commented on files/en-us/web/privacy/state_partitioning/index.md:
```diff
@@ -91,7 +91,28 @@ efforts as the Work Item is standardized.
     ["Strict" privacy protections](https://support.mozilla.org/en-US/kb/enhanced-tracking-protection-firefox-desktop#w_strict-enhanced-tracking-protection) enabled.
   - Since Firefox 90: Enabled in private browsing.
 
-### Network Partitioning
```
Trying to implement the idea from @Trikolon to have a "Static Partitioning section" with "Network Partitioning" and "Storage Partitioning"

Commit:
reverted changes to storage_access_policy

Commit:
clean up

Trikolon reviewed (commented):
Thanks for the great work! I think we can update the PR to request review now.

I'm still not super happy that now there is a bit of a disconnect between the spec talking about storage (cookies being the only storage affected) and the MDN article talking about cookies. Maybe we can discuss this with @bvandersloot-mozilla when he's back.

Commit:
add service workers to partitioned APIs

HannahPeuckmann committed:
Merge branch 'main' into always-partitioning-storage-update

HannahPeuckmann commented:
> Thanks for the great work! I think we can update the PR to request review now.
> 
> I'm still not super happy that now there is a bit of a disconnect between the spec talking about storage (cookies being the only storage affected) and the MDN article talking about cookies. Maybe we can discuss this with @bvandersloot-mozilla when he's back.

Yes, let's do so, the opinion from @bvandersloot-mozilla will be valuable!

HannahPeuckmann committed:
Merge branch 'main' into always-partitioning-storage-update

Commit:
linting

HannahPeuckmann committed:
Merge branch 'main' into always-partitioning-storage-update

HannahPeuckmann committed:
Merge branch 'main' into always-partitioning-storage-update

HannahPeuckmann committed:
Merge branch 'main' into always-partitioning-storage-update

HannahPeuckmann committed:
Merge branch 'main' into always-partitioning-storage-update

HannahPeuckmann committed:
Merge branch 'main' into always-partitioning-storage-update

dipikabh commented on files/en-us/web/privacy/state_partitioning/index.md:
```diff
@@ -85,7 +85,26 @@ efforts as the Work Item is standardized.
   - Since Firefox 90: Enabled in private browsing.
 
-### Network Partitioning
+### Static Partitioning
```
Hi @HannahPeuckmann, one feedback I have is around the section titles. We use sentence casing on MDN. So  it would be good to fix all instances.

```suggestion
### Static partitioning
```

dipikabh reviewed (changes requested):
Hi @HannahPeuckmann, I've left one feedback.

About getting more review feedback, @bvandersloot-mozilla seems to be on PTO until May (from their slack status). If there is a related spec, would it be possible to reach out to the authors/contributors for a review or with specific questions? If not, then maybe we just publish with the current changes.

teoli2003 commented on files/en-us/web/api/storage_access_api/index.md:
```diff
@@ -9,19 +9,22 @@ browser-compat:
 The semantics around third-party cookie blocking policies in particular differ from browser to browser, but the core functionality is similar: cross-origin resources embedded in a third-party context are not given access to the same cookies and site storage that they would have access to when loaded in a first-party context.
 
-These cookie blocking policies are known to break embedded cross-origin content that requires access to its first-party storage. As an example, federated logins often require access to authentication cookies stored in first-party storage, and will require the user to sign in on each site separately (or completely break) if those cookies are not available. In the case of breakage, site owners have ...
+These cookie blocking policies are known to break embedded cross-origin content that requires access to its first-party storage. As an example, federated logins often require access to authentication cookies available in a first-party context, and will require the user to sign in on each site separately (or completely break) if those cookies are not available. In the case of breakage, site owners ...
```
```suggestion
These cookie-blocking policies are known to break embedded cross-origin content that requires access to its first-party storage. As an example, federated logins often require access to authentication cookies available in a first-party context, and will require the user to sign in on each site separately (or altogether break) if those cookies are not available. In the case of breakage, site owners have often encouraged users to add their site as an exception or to disable the policy entirely. Consequently, users who wish to continue interacting with embedded content must relax significantly their blocking policy for resources loaded from all embedded origins and possibly across all websites.
```

teoli2003 commented on files/en-us/web/privacy/state_partitioning/index.md:
```diff
@@ -120,39 +139,23 @@ for cross-site tracking. As such, the following network APIs and caches are
 [Storage Access Heuristics](#storage_access_heuristics) section.
 
-#### Partitioned APIs
+#### Dynamically Partitioned APIs
```
```suggestion
#### Dynamically-partitioned APIs
```

teoli2003 commented on files/en-us/web/privacy/state_partitioning/index.md:
```diff
@@ -221,8 +219,8 @@ with storage in a third-party context. In the following examples,
 | Storage of a third-party frame is partitioned                                                           | Partitioned cookie or storage access was provided to "b.example" because it is loaded in the third-party context and storage partitioning is enabled. |
-| Storage access is granted through a **heuristic**                                                       | Storage access automatically granted for First-Party isolation "b.example" on "a.example".                                                            |
-| Storage access is granted via the [StorageAccessAPI](/en-US/docs/Web/API/Document/requestStorageAccess) | Storage access granted for origin "b.example" on "a.example".                                                                                         |
+| Access to unpartitioned cookies is granted through a **heuristic**                                                       | Storage access automatically granted for First-Party isolation "b.example" on "a.example".                                                            |
```
Can we link to the "heuristic"? Is it unique for every browser or unspecified and each can decide which one to implement?

teoli2003 commented on files/en-us/web/privacy/state_partitioning/index.md:
```diff
@@ -85,7 +85,26 @@ efforts as the Work Item is standardized.
-### Network Partitioning
+### Static Partitioning
+
+#### Storage Partitioning
```
```suggestion
#### Storage partitioning
```

teoli2003 commented on files/en-us/web/privacy/state_partitioning/index.md:
```diff
@@ -85,7 +85,26 @@ efforts as the Work Item is standardized.
+#### Storage Partitioning
+
+To prevent JavaScript accessible storage APIs being used for cross-site
+tracking, Firefox partitions accessible storage by top-level site. This
```
We should be browser-agnostic here. Or explain what the other browsers are doing, too.

teoli2003 reviewed (dismissed):
A few more comments to fix. But overall it is a significant improvement.

Commit:
less firefox centric

HannahPeuckmann committed:
Merge branch 'main' into always-partitioning-storage-update

bodhivb committed:
Fix: Use language keywords instead of framework type names in the examples of 'WebSocket' (#26725)

wbamberg committed:
Add a bit about SPAs (#26711)

dependabot[bot] committed:
Bump @mdn/yari from 2.18.1 to 2.20.0 (#26730)

rebloor committed:
DNR corrections in URLTransform's queryTransform (#26732)

oti committed:
Fix the description of the display property according to the CSS specification (#26612)

* change: `block element box` -> `block level box`

* change: `inline element boxes` -> `inline level boxes`

* change: `block element` -> `block level`

* change: `inline element` -> `inline level`

* fix: `inline/block level box` -> `inline/block box`

* change: `inline/block level` -> `inline-level/block-level element`

* fix: `block level` -> `block-level element`

yarusome committed:
Adding missing sections for `container*` (#25739)

* Update sections for `container`

* Update sections for `container-name`

* Update sections for `container-type`

* Update files/en-us/web/css/container-name/index.md

---------

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>
Co-authored-by: Estelle Weyl <estelle@weyl.org>
Co-authored-by: Brian Thomas Smith <brian@smith.berlin>

dipikabh committed:
fix(prefers-reduced-motion): Edit text and improve example (#26630)

* edits to page for clarity

* improve example explanation

* fix review comments

* Update files/en-us/web/css/@media/prefers-reduced-motion/index.md

---------

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

luckylooke committed:
Added info for 'muted' driven emission (#26641)

* Added info for 'muted' driven emission

Fixing missing information that muting also fires this event.
Standard spec: https://html.spec.whatwg.org/multipage/media.html#event-media-volumechange

* Update files/en-us/web/api/htmlmediaelement/volumechange_event/index.md

* lint

---------

Co-authored-by: wbamberg <will@bootbonnet.ca>

CanadaHonk committed:
Add page for prefers-reduced-transparency (#25289)

* Add page for prefers-reduced-transparency

* Update page for prefers-reduced-transparency

Changed text as requested in PR comments.

---------

Co-authored-by: Brian Thomas Smith <brian@smith.berlin>
Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

nalinipradash committed:
Fix arrow syntax explanation (#26704)

* remove bracket around the parameter

The example should omit the brackets around the parameter, as described in the explanation above this code.

* Update files/en-us/learn/javascript/building_blocks/functions/index.md

* Update files/en-us/learn/javascript/building_blocks/functions/index.md

* Update index.md

---------

Co-authored-by: Joshua Chen <sidachen2003@gmail.com>

hamishwillee committed:
Add RTCAudioSourceStats (#26667)

* Add RTCAudioSourceStats

* Update files/en-us/web/api/rtcaudiosourcestats/index.md

Co-authored-by: wbamberg <will@bootbonnet.ca>

* Apply suggestions from code review

Co-authored-by: wbamberg <will@bootbonnet.ca>

* Apply suggestions from code review

Co-authored-by: wbamberg <will@bootbonnet.ca>

* Update files/en-us/web/api/rtcaudiosourcestats/index.md

* Add keys for BCD data

* Make it clear that the audiolevel can also be calculated

---------

Co-authored-by: wbamberg <will@bootbonnet.ca>

yin1999 committed:
correct language tag of code fence (#26738)

frwdrik committed:
Fix code comment in dialog example (#26669)

hiddenboox committed:
Update index.md (#26744)

queengooborg committed:
Remove Prettier ignore for /web/web_components (#24225)

dependabot[bot] committed:
Bump @mdn/yari from 2.20.0 to 2.20.3 (#26747)

amirhosainshahsavari committed:
Update length values description (#26728)

* Update length values description

length values are not always absolute. For example we can use 2em for width property as a length value but it is not absolute. I corrected this in the Values section.

* Update files/en-us/web/css/width/index.md

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

---------

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

dcecile committed:
Add TransformStream note about returning a promise (#26757)

dcecile committed:
Remove emphasis on WritableStream controller (#26758)

* Remove emphasis on WritableStream controller

* copy edit

* lint

---------

Co-authored-by: wbamberg <will@bootbonnet.ca>

JasonLamv-t committed:
fix: Updated description of example which in `JavaScript.Object` to be more accurate (#26742)

Update index.md

wbamberg committed:
Fix some more return statements (#26759)

* Fix return value for toString

* Fix some more return statements

wbamberg committed:
Fix 26745: don't auto-play (#26748)

wbamberg committed:
Fix 26743: Fix return value; improve example (#26753)

wbamberg committed:
Fix return value for Range: getBoundingClientRect() (#26754)

RedBeard0531 committed:
Mention that Set iterators have a defined order (#26727)

* Mention that Set iterators have a defined order

* Do it for Map

---------

Co-authored-by: Joshua Chen <sidachen2003@gmail.com>

gkrnours committed:
Accentuate month being 0-index in getMonth example (#26734)

* Accentuate month being 0-index in get month example

`Date.getMonth` behavior differ slightly from the similar method as it return the index of the month and is 0-indexed, while the other method return ordinal values instead of index and don't have this distinction.

One could be tempted to run `${year}-${month}-${day}` and `${hour}:${minutes}:${seconds}` after the code in exemple to get iso formated date and time but the month would be off by one as iso month are 1-indexed, january being 1 and december being 12.

Adding a comment with the actual output put emphasis on this difference in behavior and might prevent people being surprised by that after the fact. The value chosen for the date is partly random. 2000 is the first year not starting with 19; January cause the month part to be 0 which should call attention to itself; 17 is high enough so it cannot be a month and I like it; 16 is in the 24h hour clock; 45 is high enough to not be confused with an hour; 30 is here for lack of imagination.

* Update files/en-us/web/javascript/reference/global_objects/date/index.md

* Update files/en-us/web/javascript/reference/global_objects/date/index.md

* Update files/en-us/web/javascript/reference/global_objects/date/index.md

---------

Co-authored-by: Joshua Chen <sidachen2003@gmail.com>

JasonLamv-t committed:
Do not use JSON.stringify in console.log (#26662)

* Update index.md

* fix: not use JSON.stringify when print via console.log

JasonLamv-t committed:
fix: Update the description of callbackFn's accumulator in `Array.reduceRight()` (#26594)

* Update index.md

* Update files/en-us/web/javascript/reference/global_objects/array/reduceright/index.md

* Update files/en-us/web/javascript/reference/global_objects/array/reduceright/index.md

---------

Co-authored-by: Joshua Chen <sidachen2003@gmail.com>

stevennyman committed:
Remove random arrow from Endianness example (#26767)

OnkarRuikar committed:
Add missed redirect (#26769)

yin1999 committed:
remove self-reference links (#26764)

* remove a self reference link

* Update index.md

postmeback committed:
Update index.md (#26771)

* Update index.md

Added the missing bracket

* Update files/en-us/web/api/stylepropertymap/append/index.md

---------

Co-authored-by: Hamish Willee <hamishwillee@gmail.com>

jasonren0403 committed:
correct http status code message (#26765)

wbamberg committed:
Fix 26741: reduce ambiguity, remove unneeded word (#26750)

* Fix 26741: reduce ambiguity, remove unneeded word

* Update files/en-us/web/http/overview/index.md

---------

Co-authored-by: Hamish Willee <hamishwillee@gmail.com>

hamishwillee committed:
Window.beforeunload event requires sticky activation (#26772)

daiwanxing committed:
Update `:paused` content (#26768)

* Update `:paused` content

* Update files/en-us/web/css/_colon_paused/index.md

* Update files/en-us/web/css/_colon_paused/index.md

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

* Update files/en-us/web/css/_colon_paused/index.md

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

* Update files/en-us/web/css/_colon_paused/index.md

Co-authored-by: A1lo <yin199909@aliyun.com>

---------

Co-authored-by: Joshua Chen <sidachen2003@gmail.com>
Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>
Co-authored-by: A1lo <yin199909@aliyun.com>

hamishwillee committed:
FF114 ECMAScript module support (#26774)

MohitMaheshwari1711 committed:
removed while loop from the example (#26588)

Co-authored-by: Mohit Maheshwari <mohitmaheshwari@1MGH-LPTP-1483.local>

HannahPeuckmann committed:
Merge branch 'main' into always-partitioning-storage-update

HannahPeuckmann committed:
clean up merge

HannahPeuckmann committed:
capitalization

dipikabh commented on files/en-us/web/api/storage_access_api/index.md:
```diff
@@ -9,19 +9,21 @@ browser-compat:
+The **Storage Access API** provides a way for embedded, cross-origin content to gain unrestricted access to the cookies that it would typically only have access to in a first-party context (We refer to storage and cookies that an origin has access to when being in a first-party context as the origin's _first-party_ storage).
 
-The API provides methods that allow embedded resources to check whether they currently have access to their first-party storage, and to request access to their first-party storage from the user agent.
+> **Note:** The [browser compatibility section](#browser_compatibility) provides an overview about the support staus of the Storage Access API.
```
This not is not needed because "Browser compatibility" is a standard section on the reference pages

```suggestion

```

HannahPeuckmann replied:
My changes to the storage_access_api doc are obsolete now, there is already an update in main that took care of the issues I am addressing in this pull request.

dipikabh commented on files/en-us/web/api/storage_access_api/index.md:
```diff
@@ -9,19 +9,21 @@ browser-compat:
 {{DefaultAPISidebar("Storage Access API")}}
 
-The Storage Access API provides a way for embedded, cross-origin content to gain unrestricted access to storage that it would normally only have access to in a first-party context (we refer to this as an origin's _first-party_ storage).
+The **Storage Access API** provides a way for embedded, cross-origin content to gain unrestricted access to the cookies that it would typically only have access to in a first-party context (We refer to storage and cookies that an origin has access to when being in a first-party context as the origin's _first-party_ storage).
```
```suggestion
The **Storage Access API** provides a way for embedded, cross-origin content to gain unrestricted access to the cookies that it would typically only have access to in a first-party context. (We refer to storage and cookies that an origin has access to when being in a first-party context as the origin's _first-party_ storage).
```

dipikabh reviewed (approved):
Thank you. It's looking good, @HannahPeuckmann, just a few nits.

Also marking my approval here.

HannahPeuckmann committed:
Merge branch 'main' into always-partitioning-storage-update

HannahPeuckmann commented on files/en-us/web/api/storage_access_api/index.md:
```diff
@@ -9,19 +9,22 @@ browser-compat:
 {{DefaultAPISidebar("Storage Access API")}}
 
-The Storage Access API provides a way for embedded, cross-origin content to gain unrestricted access to storage that it would normally only have access to in a first-party context (we refer to this as an origin's _first-party_ storage).
+The Storage Access API provides a way for embedded, cross-origin content to gain unrestricted access to the cookies that it would normally only have access to in a first-party context (We refer to storage and cookies that an origin has access to when being in a first party contex as the origin's _first-party_ storage).
```
My changes to this doc are obsolete since there is a change in main that takes care of what I have bee addressing with my changes.

HannahPeuckmann commented on files/en-us/web/privacy/state_partitioning/index.md:
```diff
@@ -120,39 +139,23 @@ for cross-site tracking. As such, the following network APIs and caches are
-
-However, unlike Network Partitioning, this boundary is dynamic and access
-to a third-party's unpartitioned storage can be granted:
+Generally Firefox partitions accessible storage by top-level site. However, this boundary is dynamic for cookies and access
```
I've tried to have it sound less Firefox-centric now. Should I also link to the compatibility section?

HannahPeuckmann commented on files/en-us/web/api/document/requeststorageaccess/index.md:
```diff
@@ -8,7 +8,10 @@ browser-compat: api.Document.requestStorageAccess
 {{APIRef("Storage Access API")}}
 
-The **`requestStorageAccess()`** method of the {{domxref("Document")}} interface returns a {{jsxref("Promise")}} that resolves if the access to first-party storage was granted, and rejects if access was denied.
+The **`requestStorageAccess()`** method of the {{domxref("Document")}} interface returns a {{jsxref("Promise")}}. If that promise resolves the requesting origin will have unrestricted access to its first-party cookies. If the access is denied though, the promise rejects.
```
Most of my changes to this doc are obsolete as well since there is a change in main that takes care of what I have been addressing with my changes.

queengooborg committed:
Merge branch 'main' into always-partitioning-storage-update

HannahPeuckmann committed:
Merge branch 'main' into always-partitioning-storage-update
